### ISSUE TITLE
The TcpReassembly class does not correctly handle the logic for closing connections with TCP RST commands.

### ISSUE THREAD

**Issue by prudens** [seladb/PcapPlusPlus#1450]
### Bug description

**Describe the bug**
During testing, I found that if the client first sends a FIN command to the server and then immediately sends an RST command, the server will immediately close the TCP connection without responding.
![image](https://github.com/seladb/PcapPlusPlus/assets/5340457/bd89c273-94d7-437e-bcb6-cd927eca3aa9)

 However, in our code logic, since the sender send the FIN command, it will enter this code branch.
```cpp
    // handle FIN/RST packets that don't contain additional TCP data
    if (isFinOrRst && tcpPayloadSize == 0)
    {
        PCPP_LOG_DEBUG("Got FIN or RST packet without data on side " << sideIndex);

        handleFinOrRst(tcpReassemblyData, sideIndex, flowKey, isRst);
        return FIN_RSTWithNoData;
    }
```
 then handleFinOrRst function will set gotFinOrRst=true.When the RST command arrives, it simply returns without handling it.
```cpp
    // if this side already got FIN or RST packet before, ignore this packet as this side is considered closed
    if (tcpReassemblyData->twoSides[sideIndex].gotFinOrRst)
    {
        PCPP_LOG_DEBUG("Got a packet after FIN or RST were already seen on this side (" << sideIndex << "). Ignoring this packet");
        return Ignore_PacketOfClosedFlow;
    }
```
As result, he other side will keep opened state forever.

**Expected behavior**
 I added some code to check for this, and it works correctly now.
```cpp
    // if this side already got FIN or RST packet before, ignore this packet as this side is considered closed
    if (twoSides_[sideIndex].gotFinOrRst)
    {
        if (!twoSides_[1 - sideIndex].gotFinOrRst && isRst)
        {
            handleFinOrRst(1 - sideIndex, isRst);
            return FIN_RSTWithNoData;
        }
        LOG_DEBUG("[{}] Got a packet after FIN or RST were already seen on this side ({}). Ignoring this packet", flowKey_, sideIndex);
        return Ignore_PacketOfClosedFlow;
    }
```

### PcapPlusPlus versions tested on

PcapPlusPlus master branch

### Operating systems tested on

Linux

### Compiler version

Clang 15

### Packet capture backend (if applicable)

libpcap

**Comment by seladb** [comment]
@prudens the purpose of TCP reassmebly is not to decide which packet to send to each side. It simply tries to reconstruct the TCP messages from a given packet flow between 2 sides.

In the case you described - once TCP reassembly sees a FIN packet, it concludes the flow has been closed, no data packets should arrive on it, and it stops processing packets on this flow. When the RST packet comes after the FIN is ignores it because the flow has been closed.

Please let me know if it makes sense, or maybe I'm missing something?

**Comment by seladb** [comment]
@prudens I think I understand now. I guess we can support this edge case. Would you consider opening a PR with a fix?
We'd also want to write a test for it, and it seems you already have an example pcap file that we can use for these tests

**Comment by seladb** [comment]
Fixed by #1496
